activerecord/lib/active_record/fixtures.rb

```diff
@@ -686,6 +694,19 @@ def check_all_foreign_keys_valid!(conn)
         def update_all_loaded_fixtures(fixtures_map) # :nodoc:
           all_loaded_fixtures.update(fixtures_map)
         end
+
+        def fixture_pk_to_label_hash # :nodoc:
+          @fixture_pk_to_label_hash ||= all_loaded_fixtures.transform_values do |fixture_set|
```

Comment:
this implementation relies on `all_loaded_fixtures` which seems to be unreliable during` fixtures_test.rb` execution.
When running `fixtures_test.rb` which contains multiple test classes. Test are executed in random order. `all_loaded_fixtures` stores the loaded fixtures of the last executed test which loaded fixtures, not the fixtures described in the class being executed by `test_label_of_composite_pk_returns_fixture_label`.